beacon-chain/blockchain/core.go

```diff
@@ -117,7 +117,12 @@ func (b *BeaconChain) CanProcessBlock(fetcher types.POWBlockFetcher, block *type
 	if mainchainBlock == nil {
 		return false, nil
 	}
-	// TODO: check if the parentHash pointed by the beacon block is in the beaconDB.
+
+	// Check if the parentHash pointed by the beacon block is in the beaconDB.
+	parentHash := block.ParentHash()
+	if _, err = b.db.Get(parentHash[:]); err != nil {
```

Comment:
So this condition is only met if there is any error. Can you also handle the non-error case where the returned value is nil? 

i.e. the database was able to execute `Get` without error, but the value never existed in the database. For leveldb, it returns an error if there is no value in the database, but this may not be the case with other database implementations.